module/VuFind/src/VuFind/Log/Logger.php

```diff
@@ -51,110 +51,243 @@
  * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
  * @link     https://vufind.org Main Site
  */
-class Logger extends BaseLogger
+
+class Logger implements LoggerInterface
 {
     /**
      * Is debug logging enabled?
      *
      * @var bool
      */
-    protected $debugNeeded = false;
+    protected bool $debugNeeded = false;
+
+    protected MonologLogger $monologLogger;
+
+    protected const LEVEL_MAP = [
+        'crit'       => LogLevel::CRITICAL,
+        'err'       => LogLevel::ERROR,
+        'warn'      => LogLevel::WARNING,
+    ];
 
     /**
-     * User IP address reader
+     * Constructor
      *
-     * @var UserIpReader
+     * @param UserIpReader       $userIpReader  User IP reader service
+     * @param MonologLogger|null $monologLogger Optional Monolog logger instance
```

Comment:
Using short form nullable type for brevity:

```suggestion
     * @param UserIpReader   $userIpReader  User IP reader service
     * @param ?MonologLogger $monologLogger Optional Monolog logger instance
```